### PR TITLE
Connect Authentication API (Login, Registration) with Frontend

### PR DESCRIPTION
**Pull request by shepherd-06**
## Features
1. Added Login, Registration API request to backend using fetch. 
2. Added cors in the backend. 
3. Added `dev` in package.json for both front-end and back-end. They use different '.env' files, that's the only difference.

These commits tie the front end with the back end. Front-end API functionalities need to be checked. I couldn't figure out how to use '.env' support on Webpack. Base API URLs can be put in the '.env' file. 

## TODO @vilbedawg 
1. Add Error handling message for `login` and `registration` API calls. 
2. API calls can be turned into a utility function. A suggestion.
3. Add `.env` support on the front end.

### CODE REVIEW

File: frontend/src/components/modules/Registeration/Registration.tsx [hunk 1]
```diff
@@ -32,15 +36,49 @@ const RegisterForm: React.FC = () => {
...
+      email: formData.email,
+      password: formData.password,
+      firstName: formData.firstname,
+      lastName: formData.lastname,
+      phoneNumber: formData.tel,
+      clubName: formData.club,
+      danRank: formData.rank,
+      underage: formData.underage,
+      guardiansEmail: formData.guardianEmail
+    };
```
Comment (vilbedawg) [inline frontend/src/components/modules/Registeration/Registration.tsx:54]: I think we could simplify this by just omitting the unneeded fields, i.e

```javascript
const { passwordConfirmation, ...data} = formData
```

**Comment by vilbedawg** [comment]
The base branch is an old feature branch and there are quite a lot of conflicts with the current development branch.

**Comment by shepherd-06** [comment]
I will add the changes from this branch on top of the new branch on a new PR.

**Comment by vilbedawg** [comment]
I will close this since these changes have been made in a separate PR.

**Pull request closed without merging**